File: grunt-src/scripts/gameSetting.js
function GameSetting() {
	function update(setting) {
		$.each(setting, function(key, value) {
			self[key](value);
		});
	}
	function toJson() {
		var ret = {};
		$.each(names, function(idx, name) {
			ret[name] = self[name]();
		});
		return ret;
	}
	function save() {
		if (localStorage) {
			localStorage.setItem("game-setting", JSON.stringify(toJson()));
		}
	}
	function load() {
		if (localStorage) {
			var str = localStorage.getItem("game-setting");
			if (str) {
				update(JSON.parse(str));
			}
		}
	}
	var self = this,
		names = [
			"fieldWidth",
			"fieldHeight",
			"pointCount",
			"wallCount",
			"codingTime",
			"gameTime",
			"turnTime"
		];
	$.each(names, function(idx, name) {
		self[name] = (function() {
			return function(v) { 
				var $el = $("#" + name);
				if (v === undefined) {
					return parseInt($el.val(), 10);
				} else {
					$el.val(v);
				}
			};
		})();
	});
	$.extend(this, {
		"update": update,
		"toJson": toJson,
		"save": save,
		"load": load
	});
}

File: app/models/GameSetting.scala
package models

import play.api.libs.json._

case class GameSetting(
  fieldWidth: Int,
  fieldHeight: Int,
  pointCount: Int,
  wallCount: Int,
  codingTime: Int,
  gameTime: Int,
  turnTime: Int
) {
  def toJson = JsObject(Seq(
    "fieldWidth" -> JsNumber(fieldWidth),
    "fieldHeight" -> JsNumber(fieldHeight),
    "pointCount" -> JsNumber(pointCount),
    "wallCount" -> JsNumber(wallCount),
    "codingTime" -> JsNumber(codingTime),
    "gameTime" -> JsNumber(gameTime),
    "turnTime" -> JsNumber(turnTime)
  ))
}

object GameSetting {
  def fromJson(json: JsValue) = {
    GameSetting(
      fieldWidth = (json \ "fieldWidth").as[Int],
      fieldHeight = (json \ "fieldHeight").as[Int],
      pointCount = (json \ "pointCount").as[Int],
      wallCount = (json \ "wallCount").as[Int],
      codingTime = (json \ "codingTime").as[Int],
      gameTime = (json \ "gameTime").as[Int],
      turnTime = (json \ "turnTime").as[Int]
    )
  }
}
